src/core_modules/capture-core/utils/cachedDataHooks/useValidatedIDsFromCache.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import { useCallback, useEffect, useState } from 'react';
+import {
+    containsKeyInStorageAsync,
+    getCachedSingleResourceFromKeyAsync,
+} from '../../MetaDataStoreUtils/MetaDataStoreUtils';
+import { userStores } from '../../storageControllers/stores';
+
+type Props = {|
+    programId: string,
+    orgUnitId?: string,
+|}
+
+const IdTypes = Object.freeze({
+    PROGRAM: 'programId',
+    ORG_UNIT: 'orgUnitId',
+});
+
+export const useValidatedIDsFromCache = ({ programId, orgUnitId }: Props) => {
+    const [valid, setValid] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState();
+
+    const getPromises = useCallback(() => {
+        const promises = [];
+        const keys = [];
+        if (programId) {
+            keys.push({ id: programId, type: IdTypes.PROGRAM });
+            promises.push(containsKeyInStorageAsync(userStores.PROGRAMS, programId));
+        }
+        if (orgUnitId) {
+            keys.push({ id: orgUnitId, type: IdTypes.ORG_UNIT });
+            promises.push(getCachedSingleResourceFromKeyAsync(userStores.ORGANISATION_UNITS_BY_PROGRAM, programId));
+        }
+
+        Promise.all(promises)
+            .then((adapterResponses) => {
+                if (orgUnitId) {
+                    adapterResponses[1] = adapterResponses[1]?.organisationUnits[orgUnitId];
```

Comment:
I am not sure that mutating the `adapterResponses` promise response is the best practice here. Maybe you can have a convertor function inside the `keys` object and apply it. Something like this:
```
keys.push({ id: orgUnitId, type: IdTypes.ORG_UNIT, convert: value => value?.organisationUnits[orgUnitId] });
keys.push({ id: programId, type: IdTypes.PROGRAM, convert: value => value });
```
And it can be used later like this
```
const adapterResponseObjects = keys.map(({ id, type, convert }, index) => ({
     id,
     valid: convert(adapterResponses[index]),
     type,
}));
```
WDYT?

Reply:
Thanks, I rewrote the hook to match your solution. Works like a charm ☺️